examples/spec/integration/reset_workflow_spec.rb

```diff
@@ -0,0 +1,163 @@
+require 'workflows/hello_world_workflow'
+require 'workflows/query_workflow'
+require 'temporal/reset_reapply_type'
+
+describe 'Temporal.reset_workflow' do
+  it 'can reset a closed workflow to the beginning' do
+    workflow_id = SecureRandom.uuid
+    original_run_id = Temporal.start_workflow(
+      HelloWorldWorkflow,
+      'Test',
+      options: { workflow_id: workflow_id }
+    )
+
+    original_result = Temporal.await_workflow_result(
+      HelloWorldWorkflow,
+      workflow_id: workflow_id,
+      run_id: original_run_id
+    )
+    expect(original_result).to eq('Hello World, Test')
+
+    new_run_id = Temporal.reset_workflow(
+      Temporal.configuration.namespace,
+      workflow_id,
+      original_run_id,
+      strategy: Temporal::ResetStrategy::FIRST_WORKFLOW_TASK
+    )
+
+    new_result = Temporal.await_workflow_result(
+      HelloWorldWorkflow,
+      workflow_id: workflow_id,
+      run_id: new_run_id,
+    )
+    expect(new_result).to eq('Hello World, Test')
+  end
+
+  def reset_hello_world_workflow_10_times(workflow_id, original_run_id, request_id:)
+    10.times.map do
```

Comment:
Is 10 really necessary? Two should catch any idempotency issues, right? Beyond that seems more like testing the Temporal server rather than the SDK.

Reply:
Fair enough, 10 was just an arbitrary number; I can replace it with 2.